### PR TITLE
fix: nft collectible improve bugs fix

### PR DESCRIPTION
**Pull request by BillySh1**
## Description

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

**Comment by BillySh1** [comment]
not completed yet

File: packages/web3-shared/base/src/specs/index.ts [hunk 1]
```diff
@@ -9,6 +9,8 @@ import type {
     ReturnNetworkResolver,
     ReturnProviderResolver,
 } from '../utils'
+export type SourceTypeOptionalRecord<T> =  { [k in SourceType]?: T }
```
Comment (guanbinrui) [inline packages/web3-shared/base/src/specs/index.ts]: I think this is not widely used. We should use it only where needs it.

File: packages/mask/src/components/shared/NFTCard/NFTInfoCard.tsx [hunk 2]
```diff
@@ -50,6 +52,12 @@ interface NFTInfoCardProps {
     sourceType?: SourceType
 }
 
+const platformCosts: SourceTypeOptionalRecord<number> = {
```
Comment (guanbinrui) [inline packages/mask/src/components/shared/NFTCard/NFTInfoCard.tsx]:
```ts
const resolvePlatformCosts = createLookupTableResolver<SourceType.OpenSea | SourceType.X2Y2 | SourceType.LooksRare, number>({
    [SourceType.OpenSea]: 2.5,
    [SourceType.X2Y2]: 0.5,
    [SourceType.LooksRare]: 2,
})
```

File: packages/mask/src/components/shared/NFTCard/NFTActivityCard.tsx [hunk 3]
```diff
@@ -99,8 +104,15 @@ export function NFTActivityCard(props: NFTActivityCardProps) {
                         {activity.to.nickname ||
                             (activity.to.address ? Others?.formatAddress(activity.to.address, 4) : '-')}
                     </strong>
-                    {formatDateTime(fromUnixTime(activity.timestamp / 1000), 'yyyy-MM-dd HH:mm')}
-                    <ExternalLink size={14} style={{ marginLeft: 4 }} />
+                    {formatDistanceToNow(activity.timestamp / 1000, {
```
Comment (guanbinrui) [inline packages/mask/src/components/shared/NFTCard/NFTActivityCard.tsx]:
```suggestion
                    {formatDistanceToNow(getUnixTime(activity.timestamp), {
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**